conformance/lang/expressions/range-expr/range_expr_exclusive.balt

```diff
@@ -0,0 +1,5847 @@
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) when value of the first expression is a Unsigned32 integer and value of the
+             second expression is a positive Signed16 integer.
+Labels: additive-expr, array-type, array:push, DecimalNumber, int, int:Signed16, int:Unsigned32, iterable-exclusive,
+        list-constructor-expr, range-expr
+
+function init() {
+    int:Unsigned32 startIndex = 1;
+    int:Signed16 endIndex = 10;
+    int count = 0;
+    int[] arr = [];
+    foreach int i in startIndex ..< endIndex {
+          arr.push(i);
+          count = count + 1;
+    }
+    io:println(count); // @output 9
+    io:println(arr); // @output [1,2,3,4,5,6,7,8,9]
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) when value of the first expression is a Unsigned32 integer and value of the
+             second expression is a positive Signed32 integer.
+Labels: additive-expr, array-type, array:push, DecimalNumber, int, int:Signed32, int:Unsigned32, iterable-exclusive,
+        list-constructor-expr, range-expr
+
+function init() {
+    int:Unsigned32 startIndex = 1;
+    int:Signed32 endIndex = 10;
+    int count = 0;
+    int[] arr = [];
+    foreach int i in startIndex ..< endIndex {
+          arr.push(i);
+          count = count + 1;
+    }
+    io:println(count); // @output 9
+    io:println(arr); // @output [1,2,3,4,5,6,7,8,9]
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) when value of the first expression is a Unsigned32 integer and value of the
+             second expression is a Unsigned8 integer.
+Labels: additive-expr, array-type, array:push, DecimalNumber, int, int:Unsigned32, int:Unsigned8, iterable-exclusive,
+        list-constructor-expr, range-expr
+
+function init() {
+    int:Unsigned32 startIndex = 1;
+    int:Unsigned8 endIndex = 10;
+    int count = 0;
+    int[] arr = [];
+    foreach int i in startIndex ..< endIndex {
+          arr.push(i);
+          count = count + 1;
+    }
+    io:println(count); // @output 9
+    io:println(arr); // @output [1,2,3,4,5,6,7,8,9]
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) when value of the first expression is a Unsigned32 integer and value of the
+             second expression is a byte.
+Labels: additive-expr, array-type, array:push, byte, DecimalNumber, int, int:Unsigned32, iterable-exclusive,
+        list-constructor-expr, range-expr
+
+function init() {
+    int:Unsigned32 startIndex = 1;
+    byte endIndex = 10;
+    int count = 0;
+    int[] arr = [];
+    foreach int i in startIndex ..< endIndex {
+          arr.push(i);
+          count = count + 1;
+    }
+    io:println(count); // @output 9
+    io:println(arr); // @output [1,2,3,4,5,6,7,8,9]
+}
+
+Test-Case: output
+Description: Test range expression (exclusive) when value of the first expression is a Unsigned32 integer and value of the
+             second expression is a Unsigned16 integer.
```

Comment:
We seem to not have tests covering,
1. both the first and second expressions belong to the byte scenario.
2. both the first and second expressions belong to singleton int. (No need to cover all combinations with one expression type being singleton I think. May be one test is enough)
3.  expressions in range-expr not being a variable ref.  e.g. simple `foreach int i in 0 ..< 3 { }` test.